docs/snippets/npm-module-linking.md

```diff
@@ -0,0 +1,214 @@
+---
+title: "Local Development of ApostropheCMS Extensions"
+detailHeading: "Tutorial"
+url: "/tutorials/local-extension-development.html"
+content: "Learn how to develop and test ApostropheCMS extensions locally without running into ES Module import resolution problems. This guide covers reliable approaches that work with Vite bundling and provides troubleshooting tips for common issues."
+tags:
+  topic: "development"
+  type: tutorial
+  effort: intermediate
+---
+# Local Development of ApostropheCMS Extensions
+
+## Why This Matters
+
+When building custom extensions for ApostropheCMS, you need a reliable way to develop and test them locally before publishing to npm. However, ES Modules in Node.js handle symlinks differently than CommonJS, and ApostropheCMS's use of Vite for bundling creates additional complexity that breaks many traditional local development approaches.
```

Comment:
I think you'll need a few sentences even more basic than this explaining what an extension is versus a project level module and that most people don't need to read this. Just so nobody gets really confused.

Reply:
Added info box and changed wording.